src/components/Page/components/Header/Header.tsx

```diff
@@ -344,3 +333,39 @@ function determineLayout({
 
   return layout.slots;
 }
+
+function getPrimaryActionMarkup(
```

Comment:
A function that returns some JSX is a React component. What do you think to naming it and treating it as such? `PrimaryActionMarkup()` and then calling it like a react component above `const primaryActionMarkup = primaryAction ? <PrimaryActionMarkup ... /> : null`

Reply:
Ah yeah this is the right decision, i'll do this.